docs/pallas/grid_blockspec.md

```diff
@@ -233,3 +236,46 @@ See [the Pallas TPU documentation](https://jax.readthedocs.io/en/latest/pallas/t
  [309 309 309 319 319 319]]
 
 ```
+
+A `None` value appearing as a dimension value in the `block_shape` behaves
+as the value `1`, except that the corresponding
+block axis is squeezed. In the example below, observe that the
+shape of the `o_ref` is (2,) when the block shape was specified as
+`(None, 2)` (the leading dimension was squeezed).
+
+```python
+>>> def kernel(o_ref):
+...   assert o_ref.shape == (2,)
+...   o_ref[...] = jnp.full((2,), 10 * pl.program_id(1) + pl.program_id(0))
+>>> pl.pallas_call(kernel,
+...                jax.ShapeDtypeStruct((3, 4), dtype=np.int32),
+...                out_specs=pl.BlockSpec((None, 2), lambda i, j: (i, j)),
+...                grid=(3, 2), interpret=True)()
+Array([[ 0,  0, 10, 10],
+       [ 1,  1, 11, 11],
+       [ 2,  2, 12, 12]], dtype=int32)
+
+```
+
+When we construct a `BlockSpec` we can use the value `None` for the
+`block_shape` parameter, in which case the shape of the overall array
+is used. And if we use the value `None` for the `index_map` parameter
```

Comment:
the shape of the entire array is used -> the entire array is used? It's not just the shape, but really the slice is the whole array